modules/juce_audio_formats/codecs/juce_WindowsMediaAudioFormat.cpp

```diff
@@ -22,6 +22,304 @@
   ==============================================================================
 */
 
+#if JUCE_MINGW
+
+} // close juce namespace
+
+#include <guiddef.h>
+#include <tuple>
+
+using	tupluuid = std::tuple<uint32_t, uint16_t, uint16_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t, uint8_t>;
+
+inline
+tupluuid	UUID_split(const char *s)
+{	
+	GUID	g = juce::uuidFromString(s);
+	
+	return std::make_tuple(g.Data1, g.Data2, g.Data3, g.Data4[0], g.Data4[1], g.Data4[2], g.Data4[3], g.Data4[4], g.Data4[5], g.Data4[6], g.Data4[7]);
+	// __CRT_UUID_DECL(name, g.Data1, g.Data2, g.Data3, g.Data4[0], g.Data4[1], g.Data4[2], g.Data4[3], g.Data4[4], g.Data4[5], g.Data4[6], g.Data4[7]);
+}
+
+/*
+struct bite
+{
+	bite(tupluuid tup)
+		: m_tupid(tup)
+	{
+	}
+	
+	auto&	operator[](const size_t idx)
+	{	return std::get<idx>(m_tupid);
+	}
+	
+	tupluuid	m_tupid;
+};
+*/
+
+/*
+#define	gh(s)	gx = UUID_split(s);
+#define	DeclUUID(typ, s)	__CRT_UUID_DECL(typ,	get<0>(gx), get<1>(gx), get<2>(gx), get<3>(gx), get<4>(gx), \
+							get<5>(gx), get<6>(gx), get<7>(gx), get<8>(gx), get<9>(gx), get<10>(gx));
+*/
+
+typedef 
+enum WMT_RIGHTS
+{
+	WMT_RIGHT_PLAYBACK			= 0x1,
+	WMT_RIGHT_COPY_TO_NON_SDMI_DEVICE	= 0x2,
+	WMT_RIGHT_COPY_TO_CD			= 0x8,
+	WMT_RIGHT_COPY_TO_SDMI_DEVICE		= 0x10,
+	WMT_RIGHT_ONE_TIME			= 0x20,
+	WMT_RIGHT_SAVE_STREAM_PROTECTED		= 0x40,
+	WMT_RIGHT_COPY				= 0x80,
+	WMT_RIGHT_COLLABORATIVE_PLAY		= 0x100,
+	WMT_RIGHT_SDMI_TRIGGER			= 0x10000,
+	WMT_RIGHT_SDMI_NOMORECOPIES		= 0x20000
+} WMT_RIGHTS;
+    
+#include "C:/Program Files (x86)/Windows Kits/8.1/Include/um/nserror.h"

```

Comment:
Many stuff has been redefined, especially the Media Format SDK stuff. But also inside the device API are types redefined which are available through MINGW MIDL or header files. I think several stuff due to historical reasons.
I agree that the marker/definition  forof feature, usage and actors(user) makes sense but the MINGW area has not been  resp. is not a focal area for past and future development. So, the effort to give mingw more coverage seems to be more driven by the engaged user base and naturaly less systematik as desired. 
So with the provisioned changes here and in the past it is sometimes not clear wether a mingw  marker/definition in code is a feature, a user or usage definition. Hard to achieve with one definition (JUCE_MING) anyway. Anyway, thanks for that nice piece of code and the opportunity to play around with it. cheers